crates/oxc_minifier/src/compressor.rs

```diff
@@ -41,25 +41,10 @@ impl<'a> Compressor<'a> {
             return;
         }
 
-        let mut i = 0;
-        loop {
-            let mut changed = false;
-            let mut pass = EarlyPass::new();
-            pass.build(program, &mut ctx);
-            if pass.changed() {
-                changed = true;
-            }
-            if !changed {
-                break;
-            }
-            if i > 50 {
-                debug_assert!(false, "Ran in a infinite loop.");
-                break;
-            }
-            i += 1;
-        }
-
-        LatePass::new().build(program, &mut ctx);
+        PeepholeOptimizations::new().build(program, &mut ctx);
+        CollapsePass::new().build(program, &mut ctx);
+        LatePeepholeOptimizations::new().run_in_loop(program, &mut ctx);
+        PeepholeOptimizations::new().build(program, &mut ctx);
```

Comment:
This is added because we changed `!1` to `bool` in late, this pass changes it back. Ideally we should remove this last pass.